### PR TITLE
[Karpenter Addon] Relaxing VPC CNI constraint and revising Instant node role outputs

### PR DESCRIPTION
**Pull request by youngjeong46**
*Issue #, if available:* #630 and #625

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by shapirov103 — COMMENTED** [review]
Lint, github action failure in index.ts.

I don't observe the fix for #630 - changing node role due to hashing applied to the token as opposed to the logical name.

File: docs/addons/karpenter.md [hunk 1]
```diff
@@ -53,12 +49,11 @@ const karpenterAddonProps = {
...
   interruptionHandling: true,
 }
-const vpcCniAddOn = new blueprints.addons.VpcCniAddOn();
+
 const karpenterAddOn = new blueprints.addons.KarpenterAddOn(karpenterAddonProps);
-const addOns: Array<blueprints.ClusterAddOn> = [ vpcCniAddOn, karpenterAddOn ];
 
 const blueprint = blueprints.EksBlueprint.builder()
-  .addOns(...addOns)
+  .addOns(...karpenterAddOn)
```
Comment (shapirov103) [inline docs/addons/karpenter.md:56]: remove ..., it is not needed for a single addon. varargs functions like addOns() allow the following:
```
.addOns(firstAddOn)
.addOns(secondAddOn)
.addOns(firstAddOn, secondAddOn)
.addOns(...arrayOfAddOns)
```

File: lib/addons/karpenter/index.ts [hunk 2]
```diff
@@ -396,20 +395,19 @@ export class KarpenterAddOn extends HelmAddOn {
             path: '/'
         });
         
-        const clusterName = cluster.clusterName;
-        const clusterHash = md5.Md5.hashStr(clusterName);
+        const clusterId = Names.uniqueId(cluster);
```
Comment (shapirov103) [inline lib/addons/karpenter/index.ts:398]: isn't that `six of one, half a dozen of the other`? cluster seems to be fairly dynamic and specific to the env, like actual build box, time, etc. 
A simple composite identifier like `uniqueId('${blueprintProps.name}-${region}'`  maybe a more straighforward choice.

**Review by shapirov103 — COMMENTED** [review]
A couple of comments.

File: docs/addons/karpenter.md (hunk 1 above)
Comment (youngjeong46) [reply docs/addons/karpenter.md:56]: addressed.

**Review by shapirov103 — APPROVED** [review]
lgtm

**Pull request merged**